static/images/diagrams/cloud_init_workflow.txt

```diff
@@ -0,0 +1,35 @@
+# Use http://sequencediagram.org/ to generate image
+
+title Cloud-based provisioning with Puppet via cloud-init
+
+participant User
+participant "Provisioned Instance" as Host
+participant "Foreman" as Foreman
+participant "Foreman Proxy" as Proxy
+participant "Infrastructure Cloud" as Cloud
+participant DNS
+participant "Puppet master" as Puppet
+
+linear
+autonumber
+
+note over Foreman:has an Image with cloud-init flag
+User->Foreman:clicks on New Host
+User->Foreman:selects cloud resource, the image and submits
+Foreman->Proxy:creates DNS records
+Proxy->DNS:forwards DNS records
+Foreman->Proxy:creates DNS records
+Foreman->Cloud:creates new instance with cloud-init info
+Foreman->Cloud:starts the instance
```

Comment:
between creating and starting the instance, it might be useful to show how the IP info comes back from the CR